### PR TITLE
Feat/1680 profile stt unapproved

### LINKED ISSUE #1680: As an unapproved grantee user, I want my requested STT in my profile
**Description:**
Since #1387 was addressed, when an unapproved grantee  user (i.e. new non-ACF user) submits a new request to TDP by submitting a request form with  their first name, last name, and STT, their profile will have a default STT value of "federal government" until their permissions group is assigned. Instead, we want the requested STT in the profile.  The current behavior for an unapproved _ACF_ user should not change. 

**Acceptance Criteria:** 
_Create a list of functional outcomes that must be achieved to complete this issue_

- [ ] Unapproved grantee user has requested STT in profile upon submitting request form
- [ ] Unapproved ACF user has "Federal Government" in profile upon submitting request form
- [ ] Unapproved grantee user profile also displays inferred region upon submitting request form
- [ ] Testing Checklist has been run and all tests pass
- [ ] README is updated, if necessary

**Tasks:**
_Create a list of granular, specific work items that must be completed to deliver the desired outcomes of this issue_ 

- [ ] Improve Profile page logic to display requested STT for grantee user
- [ ] Create backend endpoint to provide human-readable STT name to frontend if not created
- [ ] Run Testing Checklist and confirm all tests pass

**Supporting Documentation:**
_Please include any relevant log snippets/files/screen shots_
- #1387 
- _current profile for an unapproved grantee_
image

### PR DESCRIPTION
**Pull request by riatzukiza**
## Summary of Changes
_Provide a brief summary of changes_
Pull request closes #1680 

## How to Test
_List the steps to test the PR_
These steps are generic, please adjust as necessary.
```
cd tdrs-frontend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d
cd tdrs-backend && docker-compose -f docker-compose.yml -f docker-compose.local.yml up -d 
```
Create a new user and fill out the request access form, by default you should see the stt you sent in the request.
If the user has a region associated with them, the region is used instead. The backend is already enforcing this.
if there neither was submitted, then neither is set, and it will say federal government

## Tabletop

- [ ] Ready for Tabletop (Reviewer check this)
- [ ] Tabletop scheduled (Assignee check this)
- [ ] Tabletop approval 

## Deliverables
_More details on how deliverables herein are assessed included [here](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverables)._

### [Deliverable 1: Accepted Features](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-1-Accepted-Features)

Checklist of ACs:
- [x] Unapproved grantee user has requested STT in profile upon submitting request form
- [x] Unapproved ACF user has "Federal Government" in profile upon submitting request form
- [x] Unapproved grantee user profile also displays inferred region upon submitting request form
+ [x] **`lfrohlich`** and/or **`adpennington`**  confirmed that ACs are met.

### [Deliverable 2: Tested Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-2-Tested-Code)

+ Are all areas of code introduced in this PR meaningfully tested?
  + [x] If this PR introduces backend code changes, are they meaningfully tested?
  + [x] If this PR introduces frontend code changes, are they meaningfully tested?
+ Are code coverage minimums met?
  + [x] Frontend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)
  + [x] Backend coverage: [_insert coverage %_] (see `CodeCov Report` comment in PR)

### [Deliverable 3: Properly Styled Code](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-3-Properly-Styled-Code)

+ [x] Are backend code style checks passing on CircleCI?
+ [x] Are frontend code style checks passing on CircleCI?
+ [x] Are code maintainability principles being followed?

### [Deliverable 4: Accessible](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-4-Accessibility)

+ [ ] Does this PR complete the epic? 
+ [ ] Are links included to any other gov-approved PRs associated with epic?
+ [ ] Does PR include documentation for Raft's a11y review? 
+ [ ] Did automated and manual testing with `iamjolly` and `ttran-hub` using Accessibility Insights reveal any errors introduced in this PR?

### [Deliverable 5: Deployed](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-5-Deployed)

+ [x] Was the code successfully deployed via automated CircleCI process to development on Cloud.gov?

### [Deliverable 6: Documented](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-6-Code-documentation)

+ [ ] Does this PR provide background for why coding decisions were made?
+ [x] If this PR introduces backend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [x] If this PR introduces frontend code, is that code easy to understand and sufficiently documented, both inline and overall?
+ [ ] If this PR introduces dependencies, are their licenses documented?
+ [x] Can reviewer explain and take ownership of these elements presented in this code review?

### [Deliverable 7: Secure](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-7-Secure)

+ [x] Does the OWASP Scan pass on CircleCI?
+ [ ] Do manual code review and manual testing detect any new security issues?
+ [ ] If new issues detected, is investigation and/or remediation plan documented? 

### [Deliverable 8: User Research](../docs/How-We-Work/our-priorities-values-expectations.md#Deliverable-8-User-Research)

Research product(s) clearly articulate(s):
+ [ ] the purpose of the research
+ [ ] methods used to conduct the research 
+ [ ] who participated in the research
+ [ ] what was tested and how
+ [ ] impact of research on TDP
+ [ ] (_if applicable_) final design mockups produced for TDP development

### CODE REVIEW

**Review by andrew-jameson — CHANGES_REQUESTED** [review]
Deployed, not seeing Region per AC? Glad to see selected STT is correct and present though

image

**Review by reitermb — APPROVED** [review]
All looks good from an a11y standpoint! (e.g. no adverse impact to existing page nav/reading behavior)

**Comment by riatzukiza** [comment]
> Deployed, not seeing Region per AC? Glad to see selected STT is correct and present though
> 
> image

I have fixed this:
image

**Comment by andrew-jameson** [comment]
As a federal government user, I get a blank "Region" line below the rest of my profile. Is this a quick fix? @riatzukiza

File: tdrs-frontend/src/components/Profile/Profile.jsx [hunk 1]
```diff
@@ -32,23 +32,28 @@ function Profile() {
...
             } else {
-              return 'Federal Government'
+              return user?.stt?.name || user?.region?.id || 'Federal Government'
             }
           })()}
         </p>
+
+        {(() => {
+          console.log(user?.stt?.region?.id)
+          console.log(user)
```
Comment (andrew-jameson) [inline tdrs-frontend/src/components/Profile/Profile.jsx:54]: Do we want to retain these lines? Were they just for debugging?

Comment (riatzukiza) [inline tdrs-frontend/src/components/Profile/Profile.jsx:54]:
```suggestion
```

File: tdrs-frontend/src/components/Profile/Profile.jsx [hunk 2]
```diff
@@ -32,23 +32,26 @@ function Profile() {
         </div>
       )}
       <div>
-        <p className="text-bold">
+        <p id="full-name" className="text-bold">
           {user?.first_name} {user?.last_name}
         </p>
         <p>{user?.email}</p>
         <p>{primaryRole?.name}</p>
+        {(() => {
```
Comment (n0remac) [inline tdrs-frontend/src/components/Profile/Profile.jsx:40]: Might be cleaner and easier to visually parse to have this arrow function as its own function. I'd defer to someone else on that though.

**Comment by n0remac** [comment]
I don't think it was part of this ticket, but it might be worth it to change out the button link to something like:
```
           <a
              href="https://idp.int.identitysandbox.gov/account"
              target="_blank"
              rel="noopener noreferrer"
            >
              <Button
                className="usa-button"
                type="button"
                id="loginDotGovSignIn"
              >
                <div className="mobile:margin-x-auto mobile-lg:margin-0">
                  Manage Your Account at{' '}
                  <img
                    className="mobile:margin-x-auto mobile:padding-top-1 mobile-lg:margin-0 mobile-lg:padding-top-0 width-15 padding-left-1"
                    src={loginLogo}
                    alt="Login.gov"
                  />
                </div>
                <span className="visually-hidden">Opens in a new website</span>
              </Button>
            </a>
```

I noticed a couple of these onclick buttons in the code. IDK if there is a reason for them or not. I am pretty sure the hrefs are more secure and offer other features. The top answer on this SO thread explains it well https://stackoverflow.com/questions/1667416/window-location-href-vs-clicking-on-an-anchor

**Review by andrew-jameson — APPROVED** [review]
LGTM but would be interested to hear discussion around @n0remac 's suggestion on the links.

**Review by n0remac — APPROVED** [review]

**Review by ADPennington — APPROVED** [review]
non-acf user upon submitting request
1900p1

non-acf user after permissions set
1900p4

acf user upon request
1900p3

acf user after permission set
1900p5

**Pull request merged**